### ISSUE TITLE
No theme, file choser, opening links for Flatpak apps after reboot until xdg-desktop-portal is restarted

### ISSUE THREAD

**Issue by WildRikku** [ValveSoftware/SteamOS#1007]
#### Your system information

* Steam client version: steam-runtime_0.20230118.0 package version 1679680416
* SteamOS version: 3.4.6
* Opted into Steam client beta?: [Yes/No] no
* Opted into SteamOS beta?: [Yes/No] no
* Have you checked for updates in Settings > System?: [Yes/No] yes

#### Please describe your issue in as much detail as possible:
In desktop mode, apps from Flatpak, like Thunderbird, can no longer open the File Choser dialogue and can not open links in a browser and do not load themes. The issue can be fixed by `systemctl --user restart xdg-desktop-portal` except for links still not opening in the browser.

I found #775 and checked that my service unit file has the fix introduced in that issue (https://github.com/ValveSoftware/SteamOS/issues/775#issuecomment-1129427573). Installing xdg-desktop-portal-gtk as suggested in #803 is not helping and not required for the workaround.

I should note while in #775 the issue appears after switching to Steam mode and then back to Desktop mode, in my case, the issue also appears after a reboot and immediately switching to Desktop mode from the menu.

**Comment by bertogg** [comment]
Hello, the fix that from that comment that you linked is no longer necessary in SteamOS 3.4.6, I would remove `$HOME/.config/systemd/user/xdg-desktop-portal.service.d` completely and try again.

Does the problem happen always? Does it also happen if you go into Steam mode normally and then switch to Desktop mode, or only if you go straight to Desktop mode?

**Comment by WildRikku** [comment]
Thanks for your reply.

> Hello, the fix that from that comment that you linked is no longer necessary in SteamOS 3.4.6, I would remove `$HOME/.config/systemd/user/xdg-desktop-portal.service.d` completely and try again.
 
I never created that folder. What I wanted to say is that I noticed that I do not need to manually add the fix as it is already in the system's original files, just as you said.

> Does the problem happen always? Does it also happen if you go into Steam mode normally and then switch to Desktop mode, or only if you go straight to Desktop mode?

It happens in all cases that I tested. I tested:
- Boot Steam Deck from powered off / reboot, go to Steam mode, then go to desktop mode
- Go to Steam mode from desktop mode and then go back to desktop mode

You can not boot the Steam Deck directly to desktop mode (at least not without some modifications).

**Comment by bertogg** [comment]
> Boot Steam Deck from powered off / reboot, go to Steam mode, then go to desktop mode

Ok, this is quite odd... can you try this? Boot the device, switch to desktop mode then open a terminal (or log in using ssh) and type this:

`$ journalctl --user -b | grep -i portal`

**Comment by WildRikku** [comment]
> > Boot Steam Deck from powered off / reboot, go to Steam mode, then go to desktop mode
> 
> Ok, this is quite odd... can you try this? Boot the device, switch to desktop mode then open a terminal (or log in using ssh) and type this:
> 
> `$ journalctl --user -b | grep -i portal`

I rebooted then after boot I immediately switched to desktop mode and opened a terminal. The command you mentioned outputs

```
Apr 19 23:35:06 steamdeck pipewire[1259]: mod.rt: Can't find xdg-portal: (null)
Apr 19 23:35:06 steamdeck pipewire[1259]: mod.rt: found session bus but no portal
Apr 19 23:35:06 steamdeck wireplumber[1260]: Can't find xdg-portal: (null)
Apr 19 23:35:06 steamdeck wireplumber[1260]: found session bus but no portal
Apr 19 23:35:06 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Activating service name='org.freedesktop.portal.IBus' requested by ':1.9' (uid=1000 pid=1300 comm="/usr/bin/ibus-daemon -d -r --panel=disable --emoji")
Apr 19 23:35:06 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Successfully activated service 'org.freedesktop.portal.IBus'
Apr 19 23:35:18 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Activating service name='org.freedesktop.portal.IBus' requested by ':1.18' (uid=1000 pid=1779 comm="/usr/bin/ibus-daemon -d -r --panel=disable --emoji")
Apr 19 23:35:18 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Successfully activated service 'org.freedesktop.portal.IBus'
Apr 19 23:35:23 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Activating via systemd: service name='org.freedesktop.portal.Desktop' unit='xdg-desktop-portal.service' requested by ':1.21' (uid=1000 pid=1950 comm="/home/deck/.local/share/Steam/ubuntu12_32/steam -s")
Apr 19 23:35:23 steamdeck systemd[1075]: Starting Portal service...
Apr 19 23:35:23 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Activating via systemd: service name='org.freedesktop.portal.Documents' unit='xdg-document-portal.service' requested by ':1.23' (uid=1000 pid=2062 comm="/usr/lib/xdg-desktop-portal")
Apr 19 23:35:23 steamdeck systemd[1075]: Starting flatpak document portal service...
Apr 19 23:35:23 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Activating via systemd: service name='org.freedesktop.impl.portal.PermissionStore' unit='xdg-permission-store.service' requested by ':1.24' (uid=1000 pid=2072 comm="/usr/lib/xdg-document-portal")
Apr 19 23:35:23 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Successfully activated service 'org.freedesktop.impl.portal.PermissionStore'
Apr 19 23:35:23 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Successfully activated service 'org.freedesktop.portal.Documents'
Apr 19 23:35:23 steamdeck systemd[1075]: Started flatpak document portal service.
Apr 19 23:35:23 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Successfully activated service 'org.freedesktop.portal.Desktop'
Apr 19 23:35:23 steamdeck systemd[1075]: Started Portal service.
Apr 19 23:36:18 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Activating service name='org.freedesktop.portal.IBus' requested by ':1.42' (uid=1000 pid=2756 comm="/usr/bin/ibus-daemon --panel=/usr/lib/kimpanel-ibu")
Apr 19 23:36:18 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Successfully activated service 'org.freedesktop.portal.IBus'
```

In case it is of interest, here is the output that is added when I restart the service as mentioned in my original post:

```
Apr 19 23:37:32 steamdeck systemd[1075]: Stopping Portal service...
Apr 19 23:37:32 steamdeck systemd[1075]: Stopped Portal service.
Apr 19 23:37:32 steamdeck systemd[1075]: Starting Portal service...
Apr 19 23:37:32 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.gtk' unit='xdg-desktop-portal-gtk.service' requested by ':1.94' (uid=1000 pid=4077 comm="/usr/lib/xdg-desktop-portal")
Apr 19 23:37:32 steamdeck systemd[1075]: Starting Portal service (GTK/GNOME implementation)...
Apr 19 23:37:32 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Successfully activated service 'org.freedesktop.impl.portal.desktop.gtk'
Apr 19 23:37:32 steamdeck systemd[1075]: Started Portal service (GTK/GNOME implementation).
Apr 19 23:37:32 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.kde' unit='plasma-xdg-desktop-portal-kde.service' requested by ':1.94' (uid=1000 pid=4077 comm="/usr/lib/xdg-desktop-portal")
Apr 19 23:37:32 steamdeck systemd[1075]: Starting Xdg Desktop Portal For KDE...
Apr 19 23:37:32 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Activating via systemd: service name='org.freedesktop.portal.Desktop' unit='xdg-desktop-portal.service' requested by ':1.96' (uid=1000 pid=4090 comm="/usr/lib/xdg-desktop-portal-kde")
Apr 19 23:37:32 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Successfully activated service 'org.freedesktop.impl.portal.desktop.kde'
Apr 19 23:37:32 steamdeck systemd[1075]: Started Xdg Desktop Portal For KDE.
Apr 19 23:37:32 steamdeck dbus-daemon[1089]: [session uid=1000 pid=1089] Successfully activated service 'org.freedesktop.portal.Desktop'
Apr 19 23:37:32 steamdeck systemd[1075]: Started Portal service.
```

**Comment by bertogg** [comment]
Is that the complete output when you run that command **after** switching to the desktop? I see one `Started Portal service` at the end (I'm talking only about the first block that you pasted, before restarting the service), but the portal should be restarted automatically when you switch to desktop mode.

Can you run this and paste the output?

`$ systemctl --user cat xdg-desktop-portal.service`

Also this one (again, _after_ switching to desktop mode):

`$ journalctl --user -b | grep -E 'Portal|graphical user session'`

**Comment by WildRikku** [comment]
Yes, the output is complete, and yes, it is after switching to desktop mode.

The output you asked for:
```
(deck@steamdeck ~)$ journalctl --user -b | grep -E 'Portal|graphical user session'
Apr 20 21:17:09 steamdeck systemd[1070]: Reached target Current graphical user session.
Apr 20 21:17:27 steamdeck systemd[1070]: Starting Portal service...
Apr 20 21:17:27 steamdeck systemd[1070]: Started Portal service.
```
```
(deck@steamdeck ~)$ systemctl --user cat xdg-desktop-portal.service
# /usr/lib/systemd/user/xdg-desktop-portal.service
[Unit]
Description=Portal service
PartOf=graphical-session.target

[Service]
Type=dbus
BusName=org.freedesktop.portal.Desktop
ExecStart=/usr/lib/xdg-desktop-portal
Slice=session.slice
```

**Comment by bertogg** [comment]
Sorry for the late reply. There's something odd in your journalctl output, the graphical session should be restarted when you switch to desktop mode.... how do you exactly do the switch? Steam -> Power -> Switch to Desktop?

Can you provide the output of this command? (again, run it after booting the device and switching to Desktop)

`$ journalctl --user -b | grep -E 'Portal|graphical user session|Gamescope|Plasma Workspace'`

**Comment by WildRikku** [comment]
@bertogg 

> how do you exactly do the switch? Steam -> Power -> Switch to Desktop?

Yes, followed by entering the six-digit / button code.

```
(deck@steamdeck ~)$ journalctl --user -b | grep -E 'Portal|graphical user session|Gamescope|Plasma Workspace'
Apr 28 02:45:55 steamdeck systemd[1078]: Started Gamescope Session.
Apr 28 02:45:55 steamdeck systemd[1078]: Reached target Current graphical user session.
Apr 28 02:46:02 steamdeck systemd[1078]: Starting Portal service...
Apr 28 02:46:02 steamdeck systemd[1078]: Started Portal service.
Apr 28 02:46:15 steamdeck systemd[1078]: Started Gamescope Session.
Apr 28 02:46:55 steamdeck systemd[1078]: Stopping Gamescope Session...
Apr 28 02:46:55 steamdeck systemd[1078]: Stopped Gamescope Session.
Apr 28 02:46:58 steamdeck systemd[1078]: Starting KDE Plasma Workspace...
Apr 28 02:46:59 steamdeck systemd[1078]: Started KDE Plasma Workspace.
Apr 28 02:46:59 steamdeck systemd[1078]: Reached target KDE Plasma Workspace Core.
Apr 28 02:46:59 steamdeck systemd[1078]: Reached target KDE Plasma Workspace.
```

I noticed that your filter is case-sensitive. In case that was unintentional, here is the output with grep set to be case-insensitive.

```
(deck@steamdeck ~)$ journalctl --user -b | grep -Ei 'Portal|graphical user session|Gamescope|Plasma Workspace'
Apr 28 02:45:55 steamdeck systemd[1078]: Started Gamescope Session.
Apr 28 02:45:55 steamdeck systemd[1078]: Reached target Current graphical user session.
Apr 28 02:45:57 steamdeck pipewire[1264]: mod.rt: Can't find xdg-portal: (null)
Apr 28 02:45:57 steamdeck pipewire[1264]: mod.rt: found session bus but no portal
Apr 28 02:45:57 steamdeck wireplumber[1265]: Can't find xdg-portal: (null)
Apr 28 02:45:57 steamdeck wireplumber[1265]: found session bus but no portal
Apr 28 02:45:57 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Activating service name='org.freedesktop.portal.IBus' requested by ':1.10' (uid=1000 pid=1314 comm="/usr/bin/ibus-daemon -d -r --panel=disable --emoji")
Apr 28 02:45:57 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Successfully activated service 'org.freedesktop.portal.IBus'
Apr 28 02:46:02 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Activating via systemd: service name='org.freedesktop.portal.Desktop' unit='xdg-desktop-portal.service' requested by ':1.15' (uid=1000 pid=1640 comm="/home/deck/.local/share/Steam/ubuntu12_32/steam -s")
Apr 28 02:46:02 steamdeck systemd[1078]: Starting Portal service...
Apr 28 02:46:02 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Activating via systemd: service name='org.freedesktop.portal.Documents' unit='xdg-document-portal.service' requested by ':1.18' (uid=1000 pid=1775 comm="/usr/lib/xdg-desktop-portal")
Apr 28 02:46:02 steamdeck systemd[1078]: Starting flatpak document portal service...
Apr 28 02:46:02 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Activating via systemd: service name='org.freedesktop.impl.portal.PermissionStore' unit='xdg-permission-store.service' requested by ':1.19' (uid=1000 pid=1779 comm="/usr/lib/xdg-document-portal")
Apr 28 02:46:02 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Successfully activated service 'org.freedesktop.impl.portal.PermissionStore'
Apr 28 02:46:02 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Successfully activated service 'org.freedesktop.portal.Documents'
Apr 28 02:46:02 steamdeck systemd[1078]: Started flatpak document portal service.
Apr 28 02:46:02 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Successfully activated service 'org.freedesktop.portal.Desktop'
Apr 28 02:46:02 steamdeck systemd[1078]: Started Portal service.
Apr 28 02:46:09 steamdeck systemd-coredump[2159]: Process 1226 (gamescope) of user 1000 dumped core.
                                                  Module gamescope with build-id f73677b39665b78b24e46a6f1a0dec455d93cc7c
                                                  #3  0x0000560d8607a293 n/a (gamescope + 0x74293)
                                                  #4  0x0000560d86028459 n/a (gamescope + 0x22459)
                                                  #5  0x0000560d860379a1 n/a (gamescope + 0x319a1)
                                                  #1  0x0000560d8601c095 n/a (gamescope + 0x16095)
                                                  #1  0x0000560d8603a5b7 n/a (gamescope + 0x345b7)
                                                  #1  0x0000560d8604ec24 n/a (gamescope + 0x48c24)
                                                  #1  0x0000560d8603ca05 n/a (gamescope + 0x36a05)
                                                  #2  0x0000560d86016d02 n/a (gamescope + 0x10d02)
                                                  #5  0x0000560d8601a205 n/a (gamescope + 0x14205)
                                                  #3  0x0000560d86023fe5 n/a (gamescope + 0x1dfe5)
                                                  #2  0x0000560d8603ae5c n/a (gamescope + 0x34e5c)
Apr 28 02:46:14 steamdeck systemd[1078]: gamescope-session.service: Consumed 14.930s CPU time.
Apr 28 02:46:15 steamdeck systemd[1078]: Started Gamescope Session.
Apr 28 02:46:16 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Activating service name='org.freedesktop.portal.IBus' requested by ':1.30' (uid=1000 pid=2283 comm="/usr/bin/ibus-daemon -d -r --panel=disable --emoji")
Apr 28 02:46:16 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Successfully activated service 'org.freedesktop.portal.IBus'
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Failed to kill control group /user.slice/user-1000.slice/user@1000.service/app.slice/gamescope-session.service, ignoring: Operation not permitted
Apr 28 02:46:55 steamdeck systemd[1078]: Stopping Gamescope Session...
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Failed to kill control group /user.slice/user-1000.slice/user@1000.service/app.slice/gamescope-session.service, ignoring: Operation not permitted
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2235 (gamescope) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2244 (Xwayland) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2245 (Xwayland) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2281 (xbindkeys) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2283 (ibus-daemon) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2285 (power-button-ha) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2454 (steam) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2517 (steamwebhelper) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2518 (steamwebhelper) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2551 (steamwebhelper) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2809 (steamwebhelper) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2875 (steamwebhelper) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 3212 (steamos-session) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 3223 (systemctl) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2680 (steamwebhelper) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2796 (steamwebhelper) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 2529 (steamwebhelper) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Killing process 3162 (steamwebhelper) with signal SIGKILL.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Failed to kill control group /user.slice/user-1000.slice/user@1000.service/app.slice/gamescope-session.service, ignoring: Operation not permitted
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 2245 (Xwayland) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 2283 (ibus-daemon) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 2285 (power-button-ha) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 3212 (steamos-session) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 3223 (systemctl) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 2680 (steamwebhelper) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 2796 (steamwebhelper) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 2235 (gamescope) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 2244 (Xwayland) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 2454 (steam) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 2551 (steamwebhelper) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 2875 (steamwebhelper) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Unit process 2809 (steamwebhelper) remains running after unit stopped.
Apr 28 02:46:55 steamdeck systemd[1078]: Stopped Gamescope Session.
Apr 28 02:46:55 steamdeck systemd[1078]: gamescope-session.service: Consumed 1min 3.779s CPU time.
Apr 28 02:46:58 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Activating service name='org.freedesktop.portal.IBus' requested by ':1.48' (uid=1000 pid=3321 comm="/usr/bin/ibus-daemon --panel=/usr/lib/kimpanel-ibu")
Apr 28 02:46:58 steamdeck dbus-daemon[1095]: [session uid=1000 pid=1095] Successfully activated service 'org.freedesktop.portal.IBus'
Apr 28 02:46:58 steamdeck systemd[1078]: Starting KDE Plasma Workspace...
Apr 28 02:46:59 steamdeck systemd[1078]: Started KDE Plasma Workspace.
Apr 28 02:46:59 steamdeck systemd[1078]: Reached target KDE Plasma Workspace Core.
Apr 28 02:46:59 steamdeck systemd[1078]: Reached target KDE Plasma Workspace.
```

Hope that helps.

**Comment by bertogg** [comment]
Sorry again for taking so long to reply... it's interesting that Gamescope crashes in your machine:

```
Apr 28 02:46:09 steamdeck systemd-coredump[2159]: Process 1226 (gamescope) of user 1000 dumped core.
                                                  Module gamescope with build-id f73677b39665b78b24e46a6f1a0dec455d93cc7c
                                                  #3  0x0000560d8607a293 n/a (gamescope + 0x74293)
                                                  #4  0x0000560d86028459 n/a (gamescope + 0x22459)
                                                  #5  0x0000560d860379a1 n/a (gamescope + 0x319a1)
                                                  #1  0x0000560d8601c095 n/a (gamescope + 0x16095)
                                                  #1  0x0000560d8603a5b7 n/a (gamescope + 0x345b7)
                                                  #1  0x0000560d8604ec24 n/a (gamescope + 0x48c24)
                                                  #1  0x0000560d8603ca05 n/a (gamescope + 0x36a05)
                                                  #2  0x0000560d86016d02 n/a (gamescope + 0x10d02)
                                                  #5  0x0000560d8601a205 n/a (gamescope + 0x14205)
                                                  #3  0x0000560d86023fe5 n/a (gamescope + 0x1dfe5)
                                                  #2  0x0000560d8603ae5c n/a (gamescope + 0x34e5c)
```

Also, when Gamescope is stopped it should be followed by the graphical session. This is in my system:

```
May 05 15:31:52 steamdeck systemd[1152]: Stopped Gamescope Session.
May 05 15:31:52 steamdeck systemd[1152]: gamescope-session.service: Consumed 2min 6.222s CPU time.
May 05 15:31:52 steamdeck systemd[1152]: Stopped target Current graphical user session.
```

After you switch to desktop mode, what's the status of the Gamescope systemd service?

```
$ systemctl --user status gamescope-session.service 
$ systemctl --user cat gamescope-session.service 
```

**Comment by WildRikku** [comment]
I have never before heard of something like gamescope, so unfortunately, I have no idea why it behaves the way it behaves. Here is the output you asked for:

> After you switch to desktop mode, what's the status of the Gamescope systemd service?

```
(deck@steamdeck ~)$ systemctl --user status gamescope-session.service 
○ gamescope-session.service - Gamescope Session
     Loaded: loaded (/usr/lib/systemd/user/gamescope-session.service; static)
     Active: inactive (dead)

Mai 05 22:06:22 steamdeck systemd[1075]: gamescope-session.service: Unit process 2053 (steamwebhelper) remains running after unit stopped.
Mai 05 22:06:22 steamdeck systemd[1075]: gamescope-session.service: Unit process 2478 (steamwebhelper) remains running after unit stopped.
Mai 05 22:06:22 steamdeck systemd[1075]: gamescope-session.service: Unit process 2585 (steamwebhelper) remains running after unit stopped.
Mai 05 22:06:22 steamdeck systemd[1075]: gamescope-session.service: Unit process 2465 (steamwebhelper) remains running after unit stopped.
Mai 05 22:06:22 steamdeck systemd[1075]: gamescope-session.service: Unit process 1747 (Xwayland) remains running after unit stopped.
Mai 05 22:06:22 steamdeck systemd[1075]: gamescope-session.service: Unit process 2291 (steamwebhelper) remains running after unit stopped.
Mai 05 22:06:22 steamdeck systemd[1075]: gamescope-session.service: Unit process 1738 (gamescope) remains running after unit stopped.
Mai 05 22:06:22 steamdeck systemd[1075]: gamescope-session.service: Unit process 1957 (steam) remains running after unit stopped.
Mai 05 22:06:22 steamdeck systemd[1075]: Stopped Gamescope Session.
Mai 05 22:06:22 steamdeck systemd[1075]: gamescope-session.service: Consumed 51.406s CPU time.
(3)(deck@steamdeck ~)$ systemctl --user cat gamescope-session.service 
# /usr/lib/systemd/user/gamescope-session.service
[Unit]
Description=Gamescope Session
BindsTo=graphical-session.target
Before=graphical-session.target
Wants=graphical-session-pre.target
After=graphical-session-pre.target

[Service]
ExecStart=/usr/bin/gamescope-session
# This is a temporary workaround for the some proton apps becoming
# confused by the AF_UNIX socket they get for the stdout when the
# service outputs to the journal. (e.g.  Vampire Survivors).
StandardOutput=append:%t/gamescope-session.log
StandardError=append:%t/gamescope-session.log
```

**Comment by bertogg** [comment]
Gamescope is the "Steam mode". There's something weird going on because in the logs that you pasted I see that it starts twice without being stopped in the middle.

```
Apr 28 02:45:55 steamdeck systemd[1078]: Started Gamescope Session.
Apr 28 02:45:55 steamdeck systemd[1078]: Reached target Current graphical user session.
Apr 28 02:46:02 steamdeck systemd[1078]: Starting Portal service...
Apr 28 02:46:02 steamdeck systemd[1078]: Started Portal service.
Apr 28 02:46:15 steamdeck systemd[1078]: Started Gamescope Session.
```

Do you have ssh access enabled? Can you log in after booting the Steam Deck (while in Steam/Gamescope mode, before switching to the desktop) and try this?

```
$ pgrep -ax gamescope
$ systemctl --user status gamescope-session.service 
$ systemctl --user status graphical-session.target 
$ systemctl --user status xdg-desktop-portal.service 
$ systemctl --user --failed
```

**Comment by bertogg** [comment]
Also, paste the contents of `/etc/os-release`, thanks!

**Comment by WildRikku** [comment]
Hey @bertogg - sorry for the late reply, my mail server had a fault and I missed your reply.

Here comes the output - I'll enable SSH access and post again from another PC.

> Also, paste the contents of `/etc/os-release`, thanks!

```NAME="SteamOS"
PRETTY_NAME="SteamOS"
VERSION_CODENAME=holo
ID=steamos
ID_LIKE=arch
ANSI_COLOR="1;35"
HOME_URL="https://www.steampowered.com/"
DOCUMENTATION_URL="https://support.steampowered.com/"
SUPPORT_URL="https://support.steampowered.com/"
BUG_REPORT_URL="https://support.steampowered.com/"
LOGO=steamos
VARIANT_ID=steamdeck
BUILD_ID=20230313.1
VERSION_ID=3.4.6
```

**Comment by WildRikku** [comment]
```
pgrep -ax gamescope

1721 gamescope --generate-drm-mode fixed --xwayland-count 2 -w 1280 -h 800 --default-touch-mode 4 --hide-cursor-delay 3000 --max-scale 2 --fade-out-duration 200 -e -R /run/user/1000/gamescope.ca4mQEZ/startup.socket -T /run/user/1000/gamescope.ca4mQEZ/stats.pipe -O *,eDP-1 --cursor-hotspot 5,3 --cursor /usr/share/steamos/steamos-cursor.png
```
```
systemctl --user status gamescope-session.service 

● gamescope-session.service - Gamescope Session
     Loaded: loaded (/usr/lib/systemd/user/gamescope-session.service; static)
     Active: active (running) since Tue 2023-05-30 07:05:39 CEST; 1min 43s ago
   Main PID: 1698 (gamescope-sessi)
      Tasks: 232 (limit: 17686)
     Memory: 2.3G
        CPU: 1min 6.127s
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/gamescope-session.service
             ├─1698 /bin/bash /usr/bin/gamescope-session
             ├─1721 gamescope --generate-drm-mode fixed --xwayland-count 2 -w 1280 -h 800 --default-touch-mode 4 --hide-cursor-delay 3000 --max-scale 2 --fade-out-duration 200 -e -R /run/user/1000/gamescope.ca4mQEZ/startup.socket -T /run>
             ├─1730 Xwayland :0 -rootless -terminate -core -listenfd 54 -listenfd 55 -noTouchPointerEmulation
             ├─1731 Xwayland :1 -rootless -terminate -core -listenfd 59 -listenfd 61 -noTouchPointerEmulation
             ├─1767 xbindkeys -f /etc/xbindkeysrc
             ├─1769 /usr/bin/ibus-daemon -d -r --panel=disable --emoji-extension=disable
             ├─1770 /bin/bash /usr/bin/gamescope-session
             ├─1772 /usr/bin/python /usr/lib/hwsupport/power-button-handler.py
             ├─1775 mangoapp
             ├─1776 bash /home/deck/.local/share/Steam/steam.sh -steamdeck -steamos3 -steampal -steamdeck -gamepadui
             ├─1791 /usr/lib/ibus/ibus-dconf
             ├─1941 /home/deck/.local/share/Steam/ubuntu12_32/steam -steamdeck -steamos3 -steampal -steamdeck -gamepadui
             ├─1959 /home/deck/.local/share/Steam/ubuntu12_64/steamwebhelper -lang=de_DE -cachedir=/home/deck/.local/share/Steam/config/htmlcache -steampid=1941 -buildid=1682708537 -steamid=0 -logdir=/home/deck/.local/share/Steam/logs -g>
             ├─2002 "/home/deck/.local/share/Steam/ubuntu12_64/steamwebhelper --type=zygote --no-zygote-sandbox --no-sandbox --log-file=/home/deck/.local/share/Steam/logs/cef_log.txt --product-version=Valve Steam Client --lang=en_US.UTF->
             ├─2003 "/home/deck/.local/share/Steam/ubuntu12_64/steamwebhelper --type=zygote --no-sandbox --log-file=/home/deck/.local/share/Steam/logs/cef_log.txt --product-version=Valve Steam Client --lang=en_US.UTF-8"
             ├─2014 "/home/deck/.local/share/Steam/ubuntu12_64/steamwebhelper --type=gpu-process --field-trial-handle=11528292362784809759,13235224723894183789,131072 --enable-features=CastMediaRouteProvider --disable-features=SameSiteBy>
             ├─2036 "/home/deck/.local/share/Steam/ubuntu12_64/steamwebhelper --type=utility --utility-sub-type=network.mojom.NetworkService --field-trial-handle=11528292362784809759,13235224723894183789,131072 --enable-features=CastMedi>
             ├─2277 "/home/deck/.local/share/Steam/ubuntu12_64/steamwebhelper --type=renderer --no-sandbox --log-file=/home/deck/.local/share/Steam/logs/cef_log.txt --field-trial-handle=11528292362784809759,13235224723894183789,131072 -->
             ├─2393 "/home/deck/.local/share/Steam/ubuntu12_64/steamwebhelper --type=renderer --no-sandbox --log-file=/home/deck/.local/share/Steam/logs/cef_log.txt --field-trial-handle=11528292362784809759,13235224723894183789,131072 -->
             ├─2406 "/home/deck/.local/share/Steam/ubuntu12_64/steamwebhelper --type=renderer --no-sandbox --log-file=/home/deck/.local/share/Steam/logs/cef_log.txt --field-trial-handle=11528292362784809759,13235224723894183789,131072 -->
             ├─2473 "/home/deck/.local/share/Steam/ubuntu12_64/steamwebhelper --type=utility --utility-sub-type=audio.mojom.AudioService --field-trial-handle=11528292362784809759,13235224723894183789,131072 --enable-features=CastMediaRou>
             └─2597 "/home/deck/.local/share/Steam/ubuntu12_64/steamwebhelper --type=renderer --no-sandbox --force-device-scale-factor=1.50 --log-file=/home/deck/.local/share/Steam/logs/cef_log.txt --field-trial-handle=115282923627848097>

May 30 07:05:45 steamdeck pkexec[2121]: pam_unix(polkit-1:session): session opened for user root(uid=0) by (uid=1000)
May 30 07:05:45 steamdeck pkexec[2130]: pam_unix(polkit-1:session): session opened for user root(uid=0) by (uid=1000)
May 30 07:05:46 steamdeck pkexec[2157]: pam_unix(polkit-1:session): session opened for user root(uid=0) by (uid=1000)
May 30 07:05:46 steamdeck pkexec[2170]: pam_unix(polkit-1:session): session opened for user root(uid=0) by (uid=1000)
May 30 07:05:46 steamdeck pkexec[2194]: pam_unix(polkit-1:session): session opened for user root(uid=0) by (uid=1000)
May 30 07:05:46 steamdeck pkexec[2216]: pam_unix(polkit-1:session): session opened for user root(uid=0) by (uid=1000)
May 30 07:05:48 steamdeck pkexec[2331]: pam_unix(polkit-1:session): session opened for user root(uid=0) by (uid=1000)
May 30 07:05:51 steamdeck pkexec[2442]: pam_unix(polkit-1:session): session opened for user root(uid=0) by (uid=1000)
May 30 07:05:51 steamdeck pkexec[2461]: pam_unix(polkit-1:session): session opened for user root(uid=0) by (uid=1000)
May 30 07:05:55 steamdeck pkexec[2492]: pam_unix(polkit-1:session): session opened for user root(uid=0) by (uid=1000)
```
```
systemctl --user status graphical-session.target 

● graphical-session.target - Current graphical user session
     Loaded: loaded (/usr/lib/systemd/user/graphical-session.target; static)
     Active: active since Tue 2023-05-30 07:05:27 CEST; 1min 57s ago
      Until: Tue 2023-05-30 07:05:27 CEST; 1min 57s ago
       Docs: man:systemd.special(7)
```
```
systemctl --user status xdg-desktop-portal.service 

May 30 07:05:27 steamdeck systemd[1057]: Reached target Current graphical user session.
● xdg-desktop-portal.service - Portal service
     Loaded: loaded (/usr/lib/systemd/user/xdg-desktop-portal.service; static)
     Active: active (running) since Tue 2023-05-30 07:05:44 CEST; 1min 40s ago
   Main PID: 2053 (xdg-desktop-por)
      Tasks: 4 (limit: 17686)
     Memory: 2.9M
        CPU: 41ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/session.slice/xdg-desktop-portal.service
             └─2053 /usr/lib/xdg-desktop-portal

May 30 07:05:44 steamdeck systemd[1057]: Starting Portal service...
May 30 07:05:44 steamdeck xdg-desktop-por[2053]: No skeleton to export
May 30 07:05:44 steamdeck systemd[1057]: Started Portal service.
```
```
systemctl --user --failed

  UNIT                             LOAD   ACTIVE SUB    DESCRIPTION                                      
● eruption-fx-proxy.service        loaded failed failed Effects proxy daemon for Eruption
● eruption-process-monitor.service loaded failed failed Process Monitoring and Introspection for Eruption

LOAD   = Reflects whether the unit definition was properly loaded.
ACTIVE = The high-level unit activation state, i.e. generalization of SUB.
SUB    = The low-level unit activation state, values depend on unit type.
2 loaded units listed.
```

**Comment by WildRikku** [comment]
As of today, I am on SteamOS 3.4.8 and the issue seems to be resolved.

I noticed you might have found something though when you mentioned the double startup of gamescope, @bertogg. I noticed that on reboot, my Steam Deck tends to make the startup sound it makes by vibrating twice. The actual sound only plays the second time and only after the second time, I can login.

**Comment by bertogg** [comment]
I don't think there has been any change between 3.4.6 and 3.4.8 that could have had an effect on this :shrug:

Anyway, the output that you pasted seems normal to me... glad to know that it's working fine now.

**Comment by al12gamer** [comment]
I wonder if this could be effecting the issue I've been experiencing here https://github.com/ValveSoftware/SteamOS/issues/1052

**Comment by WildRikku** [comment]
> I wonder if this could be effecting the issue I've been experiencing here #1052

Hmm, it sounds similar, does my workaround work for you, too?

I do still have the issue that in some apps, I do not see my Dolphin bookmarks, but that might be unrelated.

**Comment by Konzertheld** [comment]
> Which workaround? Changing something about the gamescope service? Sorry, I get a bit lost reading here.

The one mentioned in the original post on top of this page.

**Comment by al12gamer** [comment]
No, that didn't help.

**Comment by al12gamer** [comment]
After a recent update to various flatpaks I'm no longer experiencing the https://github.com/ValveSoftware/SteamOS/issues/1052 that I referenced in here.

**Comment by WildRikku** [comment]
Ok. I will then close this ticket, as my issues also did not appear again since the above comment. I will open a new ticket for the issue with the Steam Deck apparently booting twice.

> As of today, I am on SteamOS 3.4.8 and the issue seems to be resolved.
> 
> I noticed you might have found something though when you mentioned the double startup of gamescope, @bertogg. I noticed that on reboot, my Steam Deck tends to make the startup sound it makes by vibrating twice. The actual sound only plays the second time and only after the second time, I can login. In between, there is a short "verifying installation" screen.
